### ISSUE TITLE
22.04 Deferred Improvements

### ISSUE THREAD

**Issue by jacobgkau** [pop-os/upgrade#279]
This is a list of items that were not addressed in https://github.com/pop-os/system-updater/pull/1 / https://github.com/pop-os/upgrade/pull/225 / https://github.com/pop-os/shop/pull/342, but could be improved. These can be broken out into separate issues if desired, I'm just getting them down first.

- system-updater spinboxes:
    - Make up/down keys increment spinboxes by 1 instead of 3
    - Make hours roll over when incrementing minutes past 0 or 59
    - Prevent three-digit numbers from being entered and rounding to 12
    - Allow 1 or 0 digits in the text box transiently to allow backspacing/typing more normally
- Consider making system-updater run `apt autoremove` to keep clutter from building up on the system
- Make the Pop!_Shop not immediately recheck for updates after clicking a notification from system-updater (system-updater just checked for updates, so the Pop!_Shop should be able to just use the cache)
- Fix the Pop!_Shop launching behavior so it always opens to the `Installed` tab when clicking an update notification (it currently opens to the `Home` tab if the Pop!_Shop process was already running in the background before clicking the notification)

**Comment by xploz1on** [comment]
> * Consider making system-updater run `apt autoremove` to keep clutter from building up on the system
> * Make the Pop!_Shop not immediately recheck for updates after clicking a notification from system-updater (system-updater just checked for updates, so the Pop!_Shop should be able to just use the cache)
> * Fix the Pop!_Shop launching behavior so it always opens to the `Installed` tab when clicking an update notification (it currently opens to the `Home` tab if the Pop!_Shop process was already running in the background before clicking the notification)

Pretty much sums up the "Pop!_Shop freezes" report everyone makes. Your suggestions are perfect.

**Comment by jacobgkau** [comment]
> Pretty much sums up the "Pop!_Shop freezes" report everyone makes. Your suggestions are perfect.

These have to do with [new pop-upgrade integration](https://github.com/pop-os/upgrade/pull/225) and are not related to existing Pop!_Shop performance complaints.

**Comment by xploz1on** [comment]
> * [ ]  Make the Pop!_Shop not immediately recheck for updates after clicking a notification from system-updater (system-updater just checked for updates, so the Pop!_Shop should be able to just use the cache)
 At least this one happens all the time for almost everyone. This adds the frozen stage while checking for updates. (Doesn't happen for everyone; i guess depends on internet speed?)

**Comment by jacobgkau** [comment]
> At least this one happens all the time for almost everyone. This adds the frozen stage while checking for updates.

Yes, it was expected behavior before, and that functionality being moved into a different component will have the side effect of removing it from the Pop!_Shop.
